src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -683,40 +685,55 @@ void updateInCache
   std::string geom;
   std::string coords;
   std::string georel;
+
   if (doc.hasField(CSUB_EXPR))
   {
     BSONObj expr = getObjectFieldF(doc, CSUB_EXPR);
+
     q      = expr.hasField(CSUB_EXPR_Q)?      getStringFieldF(expr, CSUB_EXPR_Q)      : "";
     mq     = expr.hasField(CSUB_EXPR_MQ)?     getStringFieldF(expr, CSUB_EXPR_MQ)     : "";
     geom   = expr.hasField(CSUB_EXPR_GEOM)?   getStringFieldF(expr, CSUB_EXPR_GEOM)   : "";
     coords = expr.hasField(CSUB_EXPR_COORDS)? getStringFieldF(expr, CSUB_EXPR_COORDS) : "";
     georel = expr.hasField(CSUB_EXPR_GEOREL)? getStringFieldF(expr, CSUB_EXPR_GEOREL) : "";
   }
 
-  int mscInsert = mongoSubCacheItemInsert(tenant.c_str(),
+  int mscInsert = mongoSubCacheItemInsert(tenant,
                                           doc,
-                                          subUp.id.c_str(),
+                                          subUp.id,
                                           servicePathCache,
                                           lastNotification,
                                           lastFailure,
                                           lastSuccess,
                                           doc.hasField(CSUB_EXPIRATION)? getLongFieldF(doc, CSUB_EXPIRATION) : 0,
                                           doc.hasField(CSUB_STATUS)? getStringFieldF(doc, CSUB_STATUS) : STATUS_ACTIVE,
-                                          q, mq, geom, coords, georel,
+                                          q,
+                                          mq,
+                                          geom,
+                                          coords,
+                                          georel,
                                           stringFilterP,
                                           mdStringFilterP,
                                           doc.hasField(CSUB_FORMAT)? stringToRenderFormat(getStringFieldF(doc, CSUB_FORMAT)) : NGSI_V2_NORMALIZED);
 
   if (mscInsert == 0)  // 0: Insertion was really made
   {
-    subCacheUpdateStatisticsIncrement();
-    if (subCacheP != NULL)
-    {
-      LM_T(LmtSubCache, ("Calling subCacheItemRemove"));
-      subCacheItemRemove(subCacheP);
-    }
+    subCacheStatisticsIncrementUpdates();
+
+    //
+    // To simulate the old way of updating:
```

Comment:
sub-cache stats were designed to help debugging the cache part. If the cache way of working has severely changed, the stats should "follow" the new implementation, i.e. maybe some fields doens't make sense any longer ans dome others need to be added.

Thus, I'd "break" legacy in this case and change it.